src/cli.rs

```diff
@@ -39,16 +41,207 @@ pub struct Config {
+            values.map(String::from).collect()
+        } else {
+            Vec::new()
+        };
+
+        let text_colors = if let Some(values) = matches.values_of("text-colors") {
+            values.map(String::from).collect()
+        } else {
+            Vec::new()
+        };
+
+        let number_of_authors: usize = matches.value_of("authors-number").unwrap().parse()?;
+
+        let ignored_directories =
+            if let Some(user_ignored_directories) = matches.values_of("exclude") {
+                user_ignored_directories.map(String::from).collect()
+            } else {
+                Vec::new()
+            };
+
+        let bot_regex_pattern = matches.is_present("no-bots").then(|| {
+            matches
+                .value_of("no-bots")
+                .map_or(Regex::from_str(r"\[bot\]").unwrap(), |s| {
+                    Regex::from_str(s).unwrap()
+                })
+        });
+
+        let language_types: Vec<LanguageType> = matches
+            .values_of("type")
+            .unwrap()
+            .map(|t| LanguageType::from_str(t).unwrap())
+            .collect();
+
+        let completion = matches
+            .value_of("completion")
+            .map(Shell::from_str)
+            .transpose()
+            .map_err(anyhow::Error::msg)
+            .context("Could not parse shell")?;
+
+        Ok(Config {
+            repo_path,
+            ascii_input,
+            ascii_language,
+            ascii_colors,
+            disabled_fields,
+            no_bold,
+            image,
+            image_backend,
+            image_color_resolution,
+            no_merges,
+            no_color_palette,
+            number_of_authors,
+            ignored_directories,
+            bot_regex_pattern,
+            print_languages,
+            print_package_managers,
+            output,
+            true_color,
+            art_off,
+            text_colors,
+            iso_time,
+            show_email,
+            include_hidden,
+            language_types,
+            completion,
+        })
+    }
+}
+
+pub fn print_supported_languages() -> Result<()> {
+    for l in Language::iter() {
+        println!("{}", l);
+    }
+
+    Ok(())
+}
+
+pub fn print_supported_package_managers() -> Result<()> {
+    for p in PackageManager::iter() {
+        println!("{}", p);
+    }
+
+    Ok(())
+}
+
+pub fn is_truecolor_terminal() -> bool {
+    env::var("COLORTERM")
+        .map(|colorterm| colorterm == "truecolor" || colorterm == "24bit")
+        .unwrap_or(false)
+}
+
+pub fn get_git_version() -> String {
+    let version = Command::new("git").arg("--version").output();
+
+    match version {
+        Ok(v) => String::from_utf8_lossy(&v.stdout).replace('\n', ""),
+        Err(_) => String::new(),
+    }
+}
+
+pub fn print_completions<G: Generator>(gen: G) {
+    let mut cmd = build_cli();
+    let name = cmd.get_name().to_string();
```

Comment:
can't we just do `let name = crate_name!();` ?

Reply:
I was mostly just following `clap_complete`'s docs, but I don't see any harm in this for our use.

Reply:
You're right, I thought we could save the call to build_cli() but `cmd` is passed to the generate function.

Sorry for the confusion

Reply:
No worries, the second call to `build_cli` in their docs looked really weird to me, too. Also that the clap command was passed as a mutable reference. I didn't look too much into it, though.